landing_page/static/landing_page/js/index.js

```diff
@@ -1,28 +1,60 @@
-function searchMovies(event) {
+async function getCSRFToken() {
+    try {
+        const response = await fetch("/get-csrf-token/");
+        const data = await response.json();
+        const csrfToken = data.csrf_token;
+
+        return csrfToken;
+    } catch (error) {
+        console.error("Error fetching CSRF token:", error);
+        throw error; // Rethrow the error to propagate it
+    }
+}
+
+
+async function searchMovies(event) {
     searchString = event.target.value.trim()
+
     // Enter key
     if (event.key === "Enter" && searchString) {
         // Route to search movies
         window.location.href = window.location.origin + "/search/movies?query=" + encodeURIComponent(searchString)
-    } else if (event.key !== "Backspace" && searchString) {
-        fetch("/", {
-            method: "POST",
-            headers: {
-                "Content-Type": "application/json"
-            },
-            body: JSON.stringify({ search: searchString })
-        })
-        .then(response => {
+    } else if (searchString) {
```

Comment:
is the point of this to, like, autofill movies?
like if i put in "batman", it'll show me
"Batman Begins"
"Batman Returns"
"Batman (1966)"
underneath the search bar?

Reply:
Yeah that was my idea